dune/gdt/discretefunction/default.hh

```diff
@@ -34,7 +91,7 @@ class ConstDiscreteFunction
                                                 typename SpaceImp::DomainFieldType, SpaceImp::dimDomain,
                                                 typename SpaceImp::RangeFieldType, SpaceImp::dimRange, SpaceImp::dimRangeCols >
 {
-  static_assert(is_space< SpaceImp >::value, "SpaceImp has to be derived from SpaceInterface!");
+//  static_assert(is_space< SpaceImp >::value, "SpaceImp has to be derived from SpaceInterface!");
```

Comment:
You could add `is_product_space` below `ProductInterface` (analogous to `is_space`) and try if
```c++
static_assert(is_space< ... >::value || is_product_space< ... >::value, "");
```
works.